tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,182 @@
+
+When we're done, the ClientUI endpoint will be sending a PlaceOrder message to itself, and then processing that message, as depicted in the following diagram:
+
+![Exercise 2 Diagram](diagram.png)
+
+
+### Create a messages assembly
+
+To share message between endpoints they need to be self-contained within a separate assembly, so let's create that assembly now.
+
+ 1. In the solution, create a new project and select the **Class Library** project type.
+ 1. Set the name of the project to **Messages**.
+ 1. Remove the automatically created **Class1.cs** file from the project. We would prefer to have our messages a little more organized than files in the root of the project. (More on this a little later.)
+ 1. Add the NServiceBus NuGet package to the Messages project.
+ 1. In the **ClientUI** project, add a reference to the **Messages** project.
+
+
+### Create a message
+
+In order to better organize messages, we'll create our first command in a folder called **Commands**.
+
+ 1. In the **Messages** project, create a new folder called **Commands**.
+ 1. In the **Commands** folder, add a new class named `PlaceOrder`.
+ 1. Mark `PlaceOrder` as `public` and implement `ICommand`.
+ 1. Add a public property of type `string` named `OrderId`.
+
+NOTE: The .NET Framework contains another interface named `ICommand` in the `System.Windows.Input` namespace. Be sure that if you use tooling to resolve the namespace, you select `NServiceBus.ICommand`. Most of the types you will need will reside in the `NServiceBus` namespace.
+
+When complete, your `PlaceOrder` class should look like the following:
+
+snippet:PlaceOrder
+
+
+### Create a handler
+
+Now that we've defined a message, we can create a corresponding message handler. Rather than create a new endpoint, let's start by just handling the message locally within the **ClientUI** endpoint.
+
+ 1. In the **ClientUI** project, create a new class named `PlaceOrderHandler`.
+ 1. Mark the handler class as public, and implement the `IHandleMessages<PlaceOrder>` interface.
+ 1. Add a logger instance, which will allow you to take advantage of the same logging system used by NServiceBus. This has an important advantage over `Console.WriteLine()`: the messages you write with the logger will be written to the log file in addition to the console. Use this code to add the logger instance to your handler class:
+    ```cs
+    static ILog logger = LogManager.GetLogger<PlaceOrderHandler>();
+    ```
+ 1. Within the `Handle` method, use the logger to record the receipt of the `PlaceOrder` message, including the value of the `OrderId` message property:
+    ```cs
+    logger.Info($"Received PlaceOrder, OrderId = {message.OrderId}");
+    ```
+ 1. Since everything we have done in this handler method is synchronous, return `Task.CompletedTask`.
+
+When complete, your `PlaceOrderHandler` class should look like this:
+
+snippet:PlaceOrderHandler
+
+
+### Send a message locally
```

Comment:
> locally

locally vs non-locally - I think that deserves some more explanation later, so maybe skip for now and explain at the end?